### ISSUE TITLE
[Bug]: multiple worker-indexes when workers=1 when using worker-scope fixtures

### ISSUE THREAD

**Issue by Danieliverant** [microsoft/playwright#33535]
### Version

1.45.0

### Steps to reproduce

I have "workers" configured to 1 in the playwright config, 
As you can see in the log PW print it's going to use only 1 worker.

I have the main fixture - "employer" - where I console.log the`test.info().workerIndex`.
Then I have another fixture(s) extends this one.
All of them are worker scope.

(adding screenshots of the above in the Additional Context section)

The tests are in separate files using the fixtures from above.

But when I print "test.info().workerIndex" in the "employer" fixture I get different indexes:

![Image](https://github.com/user-attachments/assets/499b5aad-3026-465f-89e7-101d0d6ba6c5)

(as a result of that also different employer IDs)

### Expected behavior

I expect to see only one workerIndex.

### Actual behavior

I see multiple worker indexes.

### Additional context

Using NX & nx preset (which have fullyParallel=true by default)

Employer fixture:

![Image](https://github.com/user-attachments/assets/f4ec946a-8ea2-4f4d-b39e-3189fbd338bc)

Assigned Employees fixture extending the previous one:

![Image](https://github.com/user-attachments/assets/f2edee44-87ee-446b-9c07-eedc29236d44)

Assigned employees test:

![Image](https://github.com/user-attachments/assets/7d311ab1-f560-401a-b02c-0dc672dc9778)

Employer test:

![Image](https://github.com/user-attachments/assets/fda4a8b9-70a0-42be-b058-deaaa3bfa99b)

I would expect the "employer" fixture will run only once.

### Environment

```shell
System:
    OS: macOS 14.6.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 69.67 MB / 16.00 GB
  Binaries:
    Node: 18.10.0 - ~/.nvm/versions/node/v18.10.0/bin/node
    npm: 8.19.2 - ~/.nvm/versions/node/v18.10.0/bin/npm
  Languages:
    Bash: 3.2.57 - /bin/bash
  npmPackages:
    @playwright/test: ~1.45.0 => 1.45.0
```

**Comment by pavelfeldman** [comment]
Sorry, it is hard to parse all the screenshots. Maybe https://playwright.dev/docs/test-parallel#worker-index-and-parallel-index helps?

**Comment by pavelfeldman** [comment]
Playwright will create different worker processes for tests with different worker fixtures. It'll respect --workers=1 as "have at most 1 worker at a time". But it will restart workers at will, for different fixture sets and for failing tests.

**Comment by Danieliverant** [comment]
> Playwright will create different worker processes for tests with different worker fixtures. It'll respect --workers=1 as "have at most 1 worker at a time". But it will restart workers at will, for different fixture sets and for failing tests.

even if the worker fixtures extends each other?
if B extends A, and test X uses A and test Y uses B - PW will restart the worker from test Y even though it extends fixture A?

**Comment by pavelfeldman** [comment]
Yes

**Comment by Danieliverant** [comment]
> Yes

oh thanks for approving that, I though I'm going crazy.
